### ISSUE TITLE
Potential nan problem of mean relative error loss

### ISSUE THREAD

**Issue by pengpeng-yu** [caiyuanhao1998/MST-plus-plus#50]
Hello,

Thank you for your excellent open-source work!
I’ve encountered a potential issue with the `Loss_MRAE` class when using custom datasets. Specifically, the division by label may lead to division-by-zero problems:

```python
class Loss_MRAE(nn.Module):
    def __init__(self):
        super(Loss_MRAE, self).__init__()

    def forward(self, outputs, label):
        assert outputs.shape == label.shape
        error = torch.abs(outputs - label) / label  # <- Here
        mrae = torch.mean(error.view(-1))
        return mrae
```

 What about introducing an epsilon in either the `TrainDataset` class or the `Loss_MRAE` class?

```python
class Loss_MRAE(nn.Module):
    def __init__(self):
        super(Loss_MRAE, self).__init__()

    def forward(self, outputs, label):
        assert outputs.shape == label.shape
        mask = label == 0
        if mask.any():
            label_wo_zero = label.clone()
            label_wo_zero[mask] = 1e-8
        else:
            label_wo_zero = label
        error = torch.abs(outputs - label) / label_wo_zero
        mrae = torch.mean(error)
        return mrae
``` 

Possibly related issues: #37, #49.

Thanks!

**Comment by caiyuanhao1998** [comment]
Hi thanks for reminding me,

I have updated the code.

If you find the repo useful, please help us star it, thanks :)
